packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -103,6 +123,49 @@ export function unstable_httpSubscriptionLink<
           >({
             from: eventSource,
             deserialize: transformer.output.deserialize,
+            tryHandleError: async (ev) => {
+              if (
+                'status' in ev &&
+                typeof ev.status === 'number' &&
+                typeof opts.shouldRecreateOnError === 'function' &&
+                eventSource
+              ) {
+                let recreateOnErrorOpts: RecreateOnErrorOpt = {
+                  type: 'raw',
+                  event: ev,
+                };
+                if ('status' in ev && typeof ev.status === 'number') {
+                  recreateOnErrorOpts = {
+                    type: 'http-error',
+                    status: ev.status,
+                    event: ev,
+                  };
+                }
+
+                const shouldRestart = await opts.shouldRecreateOnError(
+                  recreateOnErrorOpts,
+                );
+                if (!shouldRestart) {
+                  return false;
+                }
+
+                if (typeof opts.eventSourceOptions !== 'function') {
+                  // eslint-disable-next-line no-console
+                  console.warn(
+                    'shouldRecreateOnError has returned true but eventSourceOptions is not a function. This means eventSourceOptions will not be updated',
+                  );
+                }
+
```

Comment:
Potential race condition here -

`shouldRestart` can be called twice before it gets a chance to restart

can you make the code in a way to avoid race conditions?

Reply:
Above comment is outdated

Approach now just locks the error stream, and if an error gets handled will discard the waiting stream to avoid restarting again immediately. Feels like only compromises in this space so simple is good as a first approach I think